adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,853 @@
+                            ctx = SSL_CTX_new(TLSv1_client_method());
+                            if (ctx == NULL) {
+                                result = __LINE__;
+                                LogError("create new SSL CTX failed");
+                            }
+                            else
+                            {
+                                // LogInfo("create new SSL CTX OK");
+                                // LogInfo("SSL set fragment");
+                                /* Codes_SRS_TLSIO_SSL_ESP8266_99_086: [ If SSL_set_fragment failed, the tlsio_openssl_open shall return __LINE__. ] */
+                                ret = SSL_set_fragment(ctx, OPENSSL_FRAGMENT_SIZE);
+                                //(void*)printf("SSL_set_fragment ret:%d \n", ret);
+                                if (ret != 0){
+                                    result = __LINE__;
+                                    LogError("SSL_set_fragment failed");
+                                }
+                                else
+                                {
+                                    // LogInfo("SSL new... ");
+                                    /* Codes_SRS_TLSIO_SSL_ESP8266_99_087: [ If SSL_new failed, the tlsio_openssl_open shall return __LINE__. ] */
+                                    ssl = SSL_new(ctx);
+                                    //(void*)printf("after ssl new \n");
+                                    if (ssl == NULL) {
+                                        result = __LINE__;
+                                        LogError("create ssl failed");
+                                    }
+                                    else
+                                    {
+                                        // LogInfo("SSL set fd");
+                                        // returns 1 on success
+                                        /* Codes_SRS_TLSIO_SSL_ESP8266_99_088: [ If SSL_set_fd failed, the tlsio_openssl_open shall return __LINE__. ] */
+                                        ret = SSL_set_fd(ssl, sock);
+                                        //(void*)printf("SSL_set_fd ret:%d \n", ret);
+                                        if (ret != 1){
+                                            result = __LINE__;
+                                            LogError("SSL_set_fd failed");
+                                        }
+                                        else{
+                                            // LogInfo("SSL connect... ");
+                                            /* Codes_SRS_TLSIO_SSL_ESP8266_99_027: [ The tlsio_openssl_open shall set the tlsio to try to open the connection for 20 times before assuming that connection failed. ]*/
+                                            /* Codes_SRS_TLSIO_SSL_ESP8266_99_089: [ If SSL_connect failed, the tlsio_openssl_open shall return __LINE__. ] */
+                                            int retry = 0;
```

Comment:
Please change the `retry` name for something else, you already have another variable with the same name in this context.